### ISSUE TITLE
fix(tab): fix and document forwardRef type of Tab

### ISSUE THREAD

**Issue by SpyTec** [patternfly/patternfly-react#2773]
Right now if you try to use Tabs or Tab, you will not get proper TypeScript linting errors for entering in wrong prop types

E.g.
```jsx
<Tabs activekey={this.state.activeTabKey} onSelect={this.handleTabClick}>
  <Tab eventKey={0} title={null}>content</Tab>
  <Tab eventKey={null} title={<div>test</div>}>content</Tab>
</Tabs>
```

Here, neither will report an issue with TypeScript

**Comment by SpyTec** [comment]
My guess is that the props somehow turn into `any`. I couldn't figure out how to fix it without removing `withForwardedRef()`

Linting works for everything but Tab so far.

## Expected results
![image](https://user-images.githubusercontent.com/6598829/63701280-f2d60180-c824-11e9-9093-2bc48795dd60.png)


## Actual results
![image](https://user-images.githubusercontent.com/6598829/63701252-e5207c00-c824-11e9-8939-bf837d8a8581.png)

**Comment by SpyTec** [comment]
@rachael-phillips this isn't blocking me :)

**Comment by boaz0** [comment]
@SpyTec is this still happening in the new release?

**Comment by SpyTec** [comment]
On v3.112.3 it is still happening for me

**Comment by redallen** [comment]
@rachael-phillips No, #2718 wasn't the fix. Now [our Tab component](https://github.com/patternfly/patternfly-react/blob/master/packages/patternfly-4/react-core/src/components/Tabs/Tab.tsx) looks kinda funny, though. Not sure if it's been released yet, can you try version 3.117.3 @SpyTec ?

**Comment by SpyTec** [comment]
Not able to update yet, deps not released :)
```
No matching version found for @patternfly/react-icons@^3.14.13.
```
